src/js/angular/models/import/import-resource-tree-element.js

```diff
@@ -0,0 +1,153 @@
+     *
+     * @return {ImportResourceTreeElement} the found or newly created resource.
+     */
+    getOrCreateDirectory(directoryName) {
+        let directory = this.directories.find((directory) => directory.name === directoryName);
+        if (!directory) {
+            directory = new ImportResourceTreeElement();
+            directory.name = directoryName;
+            this.directories.push(directory);
+        }
+        return directory;
+    }
+
+    /**
+     * @return {ImportResourceTreeElement} - the root resource of the paren-child tree.
+     */
+    getRoot() {
+        if (this.parent === undefined) {
+            return this;
+        }
+        return this.getRoot(this.parent);
+    }
+
+    /**
+     * Checks if current {@link ImportResourceTreeElement} is container of other resources.
+     *
+     * @return {boolean} true if the resource contains other resources.
+     */
+    isDirectory() {
+        return ImportResourceType.DIRECTORY === this.importResource.type;
+    }
+
+    /**
+     * Checks if current {@link ImportResourceTreeElement} has rdf data.
+     *
+     * @return {boolean} true if the resource contains rdf data.
+     */
+    isFile() {
+        return ImportResourceType.FILE === this.importResource.type;
+    }
+
+    /**
+     * Checks if resource contains other resources.
+     *
+     * @return {boolean} true if resource contain rdf resources.
+     */
+    isEmpty() {
+        return this.directories.length === 0 && this.files.length === 0;
+    }
+
+    /**
+     * Returns flattening view of resources parent-child relation tree.
+     *
+     * @return {ImportResourceTreeElement[]} flattening view of resources.
+     */
+    toList() {
+        const result = [];
+        if (this.parent) {
+            // skip the root element.
+            result.push(this);
+        }
+        this.directories.forEach((directory) => {
+            result.push(...directory.toList());
+        });
+        result.push(...this.files);
+        return result;
+    }
+
+    getParentsCount() {
+        let parentCount = 0;
+        if (this.parent === undefined) {
+            return parentCount;
+        }
+
+        let parent = this.parent;
+        while (parent) {
+            parent = parent.parent;
+            parentCount++;
+        }
+        return parentCount - 1;
+    }
+
+    setSelection(selected) {
+        this.selected = selected;
+        this.files.forEach((file) => file.setSelection(selected));
+        this.directories.forEach((directory) => directory.setSelection(selected));
+    }
+
+    getAllSelected() {
+        const allSelected = [];
+        if (this.selected) {
+            allSelected.push(this);
+        }
+
+        this.files.forEach((file) => allSelected.push(...file.getAllSelected()));
+        this.directories.forEach((directory) => allSelected.push(...directory.getAllSelected()));
+
+        return allSelected;
+    }
+
+    getResourceName(resourceName) {
```

Comment:
This actually returns the resource and not its name. Better rename it to `getResourceByName` for example.

Reply:
Done